src/Atmos/Parameterizations/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -45,146 +42,165 @@ module SurfaceFluxes
+                       z_0_m::DT,
+                       z_0_h::DT,
+                       z::DT,
+                       F_m::DT,
+                       F_h::DT,
+                       a::DT,
+                       Pr::DT,
+                       pottemp_flux::DT
+                       ) where DT<:AbstractFloat
+
+Surface conditions given
+ - `u_ave` volume-averaged horizontal wind speed
+ - `θ_ave` volume-averaged potential temperature
+ - `θ_bar` basic potential temperature
+ - `θ_s` surface potential temperature
+ - `Δz` layer thickness (not spatial discretization)
+ - `z_0_m` roughness length for momentum
+ - `z_0_h` roughness length for heat
+ - `z` coordinate axis
+ - `F_m` momentum flux at the top
+ - `F_h` heat flux at the top
+ - `a` free model parameter with prescribed value of 4.7
+ - `Pr` Prantl number at neutral stratification
+ - `pottemp_flux` potential temperature flux (optional)
+
+If `pottemp_flux` is not given, then it is computed by iteration
+of equations 3, 17, and 18 in Nishizawa2018.
 """
-    compute_friction_velocity(u_ave, flux, z_0, z_1, β_m, γ_m, tol_abs, iter_max)
+function surface_conditions(u_ave::DT,
+                            θ_bar::DT,
+                            θ_ave::DT,
+                            θ_s::DT,
+                            Δz::DT,
+                            z_0_m::DT,
+                            z_0_h::DT,
+                            z::DT,
+                            F_m::DT,
+                            F_h::DT,
+                            a::DT,
+                            Pr::DT,
+                            pottemp_flux::Union{Nothing, DT}=nothing
+                            ) where DT<:AbstractFloat
+
+  L_MO = 1.0
+  if pottemp_flux!=nothing
+
+    @assert 0 <= z <= z_0_m
+    @assert 0 <= z <= z_0_h
+
+    xatol, maxiters = DT(1e-3), 100
+    u_star = compute_friction_velocity(Δz, a, xatol, maxiters, θ_bar, pottemp_flux, z_0_m, u_ave, 1)
+    θ_star = compute_temperature_scale(Δz, a, xatol, maxiters, θ_bar, pottemp_flux, z_0_h, θ_ave, θ_s, 1/Pr, u_star)
+    pottemp_flux_star = pottemp_flux
 
-Computes roots of friction velocity equation (Eq. 10 in Ref. Byun1990)
+  else
 
-`u_ave = u_* ( ln(z/z_0) - ψ_m(z/L, z_0/L) ) /κ        Eq. 10 in Ref. Byun1990`
-"""
-function compute_friction_velocity(u_ave, flux, z_0, z_1, β_m, γ_m, tol_abs, iter_max)
-
-  ustar_0 = u_ave * k_Karman / log(z_1 / z_0)
-  ustar = ustar_0
-  let u_ave=u_ave, flux=flux, z_0=z_0, z_1=z_1, β_m=β_m, γ_m=γ_m
-
-    # use neutral condition as first guess
-    stable = z_1/compute_MO_len(ustar_0, flux) >= 0
-    function compute_ψ_m(u)
-      L_MO = compute_MO_len(u, flux)
-      ζ   = z_1/L_MO
-      ζ_0 = z_0/L_MO
-      return stable ? ψ_m_stable(ζ, ζ_0, β_m) : ψ_m_unstable(ζ, ζ_0, γ_m)
-    end
-    function compute_u_ave_over_ustar(u)
-      return (log(z_1 / z_0) - compute_ψ_m(u)) / k_Karman # Eq. 10 in Ref. Byun1990
+    initial_x = [100, 15.0, 350.0] # TODO: improve initial guess
```

Comment:
Use values from previous time step by default?

Reply:
Agreed, I'll pass this as an input argument.